File: Objective-C Demo App/iOS-SDK-Demo/ALDemoBaseViewController.m
//
//  ALDemoBaseViewController.m
//  iOS-SDK-Demo
//
//  Created by Thomas So on 9/23/15.
//  Copyright © 2015 AppLovin. All rights reserved.
//

#import "ALDemoBaseViewController.h"

@implementation ALDemoBaseViewController

#pragma mark - View Lifecycle

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear: animated];
    [self.navigationController setToolbarHidden: NO animated: YES];
}

- (void)viewWillDisappear:(BOOL)animated
{
    [self.navigationController setToolbarHidden: YES];
    [super viewWillDisappear: animated];
}


#pragma mark - Logging

- (void)log:(NSString *)message
{
    dispatch_async(dispatch_get_main_queue(), ^{
        if ( self.adStatusLabel )
        {
            self.adStatusLabel.text = message;
        }
        ALLog(@"[%@] : %@", NSStringFromClass([self class]), message);        
    });
}

@end

File: Swift Demo App/iOS-SDK-Demo/ALDemoBaseViewController.swift
//
//  ALDemoBaseViewController.swift
//  iOS-SDK-Demo
//
//  Created by Thomas So on 9/25/15.
//  Copyright © 2015 AppLovin. All rights reserved.
//

import UIKit

class ALDemoBaseViewController: UIViewController
{
    @IBOutlet weak var adStatusLabel: UILabel?
    
    override func viewDidAppear(_ animated: Bool)
    {
        super.viewDidAppear(animated)
        self.navigationController?.setToolbarHidden(self.hidesBottomBarWhenPushed, animated: true)
    }
    
    override func viewWillDisappear(_ animated: Bool)
    {
        self.navigationController?.setToolbarHidden(true, animated: false)
        super.viewWillDisappear(animated)
    }
    
    func log(_ message: String!)
    {
        DispatchQueue.main.async {
            if let label = self.adStatusLabel {
                label.text = message
            }
        };
    }
}
